docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+
+We also mention, when relevant, what is stored in the database (as attributes, so that it can be easily queried e.g. with the :ref:`QueryBuilder <how-to:data:find>`) and what is stored in the file repository (providing access to the file contents, but not efficiently queryable: this is useful for e.g. big data files that don't need to be queried for).
+
+For all data types, you can follow the link to the data class to read more about the methods provided, how to access them, and so on.
+
+If you need to work with some specific type of data, first check the list of data types/plugins below, and if you don't find what you need, give a look to :ref:`how to write a new data plugin <how-to:data:plugin:create>`.
+
+.. _topics:data_types:core:base:
+
+Base types
+----------
+
+In the :py:mod:`aiida.orm.nodes.data.base` module there are a number of useful classes that wrap base python datatypes (like :py:class:`~aiida.orm.nodes.data.int.Int`, :py:class:`~aiida.orm.nodes.data.float.Float`, :py:class:`~aiida.orm.nodes.data.str.Str`, ...).
+These are automatically loaded with the verdi shell, and also directly exposed from :py:mod:`aiida.orm`.
+These classes are particularly useful when you need to provide a single parameter to e.g. a :py:class:`~aiida.engine.processes.functions.workfunction`.
+
+Each of these classes can most often be used transparently (e.g. you can sum two :py:class:`~aiida.orm.nodes.data.int.Int` objects, etc.). If you need to access the bare value and not the whole AiiDA class, use the ``.value`` property.
+
+In the same module, there is also a :py:class:`~aiida.orm.nodes.data.list.List` class to store a list of base data types.
+
+The different datatypes can be accessed through the :py:func:`~aiida.plugins.factories.DataFactory` function (also exposed from :py:mod:`aiida.plugins`) by passing an entry point to it as an argument.
+A list of all the data entry points can be obtain running the command ``verdi plugin list aiida.data``.
+
+.. _topics:data_types:core:dict:
+
+Dict
+----
+
+* **Class**: :py:class:`~aiida.orm.nodes.data.dict.Dict`
+* **String to pass to the** :py:func:`~aiida.plugins.factories.DataFactory`: ``dict``
+* **Aim**: store a dictionary of python base types in the database.
+  It can store any dictionary where elements can be a base python type (strings, floats, integers, booleans, None type, datetime objects) and lists or dictionaries of them, at any depth level (e.g. a dictionary where a value is a list of dictionaries of strings and floats).
+* **What is stored in the database**: all key/values pairs as attributes
+* **What is stored in the file repository**: ---
+
+.. _topics:data_types:core:array:
+
+ArrayData
+---------
+
+* **Class**: :py:class:`~aiida.orm.nodes.data.array.ArrayData`
+* **String to pass to the** :py:func:`~aiida.plugins.factories.DataFactory`: ``array``
+* **Aim**: store generic numeric arrays
+* **What is stored in the database**: the shape of the arrays and the name of the arrays
```

Comment:
Probably here an example would be useful, and also in some below (like SinglefileData) to see what the keys are